acinclude/squid-util.m4

```diff
@@ -332,30 +332,3 @@ AS_IF([test "$ac_res" != no],[
   $3],[$4])
 AS_VAR_POPDEF([ac_Search])dnl
 ])
-
-dnl Check for Cyrus SASL
-AC_DEFUN([SQUID_CHECK_SASL],[
-  squid_cv_check_sasl="auto"
-  AC_CHECK_HEADERS([sasl/sasl.h sasl.h])
-  AC_CHECK_LIB(sasl2,sasl_errstring,[LIBSASL="-lsasl2"],[
-    AC_CHECK_LIB(sasl,sasl_errstring,[LIBSASL="-lsasl"], [
-      squid_cv_check_sasl="no"
-    ])
-  ])
-  AS_IF([test "$squid_host_os" = "darwin"],[
-    AS_IF([test "$ac_cv_lib_sasl2_sasl_errstring" = "yes"],[
-      AC_DEFINE(HAVE_SASL_DARWIN,1,[Define to 1 if Mac Darwin without sasl.h])
```

Comment:
This `AC_DEFINE(HAVE_SASL_DARWIN` being removed should be matched with removal of all uses of the `HAVE_SASL_DARWIN` in the code. Most look trivial and already pointless.

Though there is one `#elif HAVE_SASL_DARWIN` in `src/acl/external/kerberos_ldap_group/support_sasl.cc` which will need a closer look and a test on MacOS that the helper still works after this PR.

Reply:
I can check it builds, I can't runtime test it, but given that after this we're relying on Cyrus SASL and not on the Mac one, it should work just like everywhere else provided that it builds

Reply:
Confirmed building